gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/javaversions/JdkManager.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * (c) Copyright 2022 Palantir Technologies Inc. All rights reserved.&#10;&#10;Licensed under the Apache License, Version 2.0 (the &quot;License&quot;);&#10;you may not use this file except in compliance with the License.&#10;You may obtain a copy of the License at&#10;&#10;    http://www.apache.org/licenses/LICENSE-2.0&#10;&#10;Unless required by applicable law or agreed to in writing, software&#10;distributed under the License is distributed on an &quot;AS IS&quot; BASIS,&#10;WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.&#10;See the License for the specific language governing permissions and&#10;limitations under the License.
+ */
+
+package com.palantir.baseline.plugins.javaversions;
+
+import java.io.IOException;
+import java.nio.file.FileAlreadyExistsException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.util.UUID;
+import java.util.stream.Stream;
+import org.apache.commons.io.FileUtils;
+import org.rauschig.jarchivelib.Archiver;
+import org.rauschig.jarchivelib.ArchiverFactory;
+
+public final class JdkManager {
+    private final Path storageLocation;
+    private final AzulJdkDownloader azulJdkDownloader;
+
+    JdkManager(Path storageLocation, AzulJdkDownloader azulJdkDownloader) {
+        this.storageLocation = storageLocation;
+        this.azulJdkDownloader = azulJdkDownloader;
+    }
+
+    public Path jdk(JdkSpec jdkSpec) {
+        Path jdkPath = storageLocation.resolve(String.format(
+                "zulu-%s-%s-%s",
+                jdkSpec.javaVersion(), jdkSpec.zuluVersion(), jdkSpec.hash().substring(0, 8)));
+
+        if (Files.exists(jdkPath)) {
+            return jdkPath;
+        }
+
+        Path jdkArchive = azulJdkDownloader.downloadJdkFor(jdkSpec);
+
+        Archiver archiver = ArchiverFactory.createArchiver(jdkArchive.toFile());
+        Path temporaryJdkPath = Paths.get(
+                jdkPath + ".in-progress-" + UUID.randomUUID().toString().substring(0, 8));
+        try {
+            try {
+                archiver.extract(jdkArchive.toFile(), temporaryJdkPath.toFile());
+            } catch (IOException e) {
+                throw new RuntimeException("Failed to extract jdk to directory", e);
+            }
```

Comment:
I wonder if it would be possible/beneficial to leverage [gradle cacheable artifact transforms](https://docs.gradle.org/current/userguide/artifact_transforms.html)?
That way we could reuse the same unpacked jdk across projects/checkouts instead of extracting in each repo.

Reply:
I was actually just going to extract to `~/.gradle/gradle-baseline/jdks` to reuse jdks across repos and depend on the using a hash to keep them the same - but this is worth investigating, particularly as I wouldn't have to write all this caching code myself!